### PR TITLE
modrelu activation function

### PR DESCRIPTION
**Pull request by jingli9111**
modrelu is widely used in complex domain. 
It has been proved to be useful in several models:
https://arxiv.org/abs/1511.06464, https://arxiv.org/abs/1612.05231

The implementation calculate the magnitude and phases of complex numbers then apply biased relu on the magnitude.

### CODE REVIEW

**Comment by vrv** [comment]
Overall this looks fine, and since you're the author of the papers listed I presume you are implementing it to your spec, but can you take some time to look at https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md#contribution-guidelines-and-standards to make sure this PR matches the style of all other TensorFlow python code?  There's a lot that would likely fail our sanity check linter.

You will also want to expose the symbol in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/nn/__init__.py so it will be available as tf.contrib.nn.modrelu.

**Comment by jingli9111** [comment]
Hi,

I have updated the code according to the style document. It's also exposed in the __init__.

File: tensorflow/contrib/nn/python/ops/modrelu_test.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for modrelu.py."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import modrelu
+# from tensorflow.contrib.nn.python.ops import modrelu
```
Comment (vrv) [inline tensorflow/contrib/nn/python/ops/modrelu_test.py]: Our style is to import the full module paths - is there any reason it didnt work?

Comment (jingli9111) [inline tensorflow/contrib/nn/python/ops/modrelu_test.py]: I think it should work. I will change it to full path.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by jingli9111** [comment]
Looks like the test fails. You are right. There are some problems with the import. 
It cannot do `from tensorflow.contrib.nn.python.ops import modrelu`
Is it because in the code I put `from tensorflow.python.ops import nn_ops` which creates some circular dependency? How can I fix it?

**Review by vrv — APPROVED** [review]

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@jingli9111 please resolve test errors.

**Comment by vrv** [comment]
yeah the problem appears to be:

```
ImportError: No module named 'tensorflow.contrib.nn.python.ops.mod_relu'
```

so that needs to be fixed.

once you rebase / fix the test, can you confirm by running a few tests locally and let us know if it's passing?  Thanks!

**Comment by vrv** [comment]
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/nn/BUILD You also need to add the files and the test declaration to the BUILD file like the other operations.

**Comment by jingli9111** [comment]
Thanks! I will fix it soon!

**Comment by martinwicke** [comment]
@jingli9111 any news?

**Comment by drpngx** [comment]
@jingli9111 feel free to also pull rebase and push again. Thanks!

**Comment by sb2nov** [comment]
@jingli9111 ping

**Comment by martinwicke** [comment]
I will close this PR for inactivity. I can reopen after you fix the conflicts and the BUILD, or you can make a new PR.

**Pull request closed without merging**